### PR TITLE
BP-440: Implemented settings view

### PR DESCRIPTION
**Pull request by julianweise**

### CODE REVIEW

File: src/SettingsView/SettingsView.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+						eventType={eventType} 
+						key={eventType.Id}
+						deleteEventType={this.props.deleteEventType} />);
+				})}
+			</div>
+		);
+	}
+}
+
+export default connect.defaults({fetch: ConnectionComponent.switchFetch})(props => ({
```
Comment (hpibptlab) [inline src/SettingsView/SettingsView.js]: MAJOR Remove the unused function parameter "props". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/06a7ca2353680a1e05e0c112bd401ceab2382ba1/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/dfa66769f446923b1fddb031250f4806e499f27e/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/e4ea5ea29c73681353ec0f1ef916359f6a8fea1b/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/d06e192d4867602c29f1e2d5825cee3bff276ccd/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/85f66ad727df1fc7edfb90d28d39f7196a3399ad/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

File: src/SettingsView/EventType.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import {List, ListItem} from 'material-ui/List';
+import EventQueryListItem from  './EventQueryListItem';
+import config from './../config/config.js';
+import { css } from 'aphrodite';
+import AppStyles from '../AppStyles';
+import {connect, PromiseState} from 'react-refetch';
+import ConnectionComponent from './../Utils/ConnectionComponent.js';
+
+
+class EventType extends ConnectionComponent {
```
Comment (Munin33) [inline src/SettingsView/EventType.js:12]: throws Typeerrors

File: src/SettingsView/EventType.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		if(connectionIncomplete) {
+			return connectionIncomplete;
+		}
+		return (
+			<Card
+				expanded={this.state.expanded}
+				onExpandChange={this.handleExpandChange}>
+				<CardHeader
+					title={this.props.eventType.Name}
+					subtitle={`${config.textNumberOfEvents} ${this.props.eventType.NumberOfEvents}`}
```
Comment (Munin33) [inline src/SettingsView/EventType.js]: can we do this better? kinda hacky?

File: src/SettingsView/EventType.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+					actAsExpander={true}
+					showExpandableButton={true}
+				/>
+				<CardText
+					expandable={true}
+					className={css(AppStyles.dFlex)}>
+					<List className={css(AppStyles.w50)}>
+						{queries.forEach((query) => {
+								return(<EventQueryListItem
```
Comment (Munin33) [inline src/SettingsView/EventType.js]: break this line please

File: src/SettingsView/SettingsView.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	render() {
+		const connectionIncomplete = super.render(this.props.eventTypes);
+		if(connectionIncomplete) {
+			return connectionIncomplete;
+		}
+		return (
+			<div>
+				<SearchBar/>
+				{this.props.eventTypes.value.map((eventType) => {
+					return(<EventType
```
Comment (Munin33) [inline src/SettingsView/SettingsView.js:19]: break this line please

File: src/SettingsView/SettingsView.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	eventTypes: `/eventtypes`,
+	deleteEventType: eventType => ({
+		postLikeResponse: {
+			url: `/eventtype/${eventType.Id}/delete`,
+			method: 'DELETE',
+			body: "",
+			andThen: () => ({
+				eventTypes: {
+					url: `/eventtypes`,
+					refreishing: true
```
Comment (Munin33) [inline src/SettingsView/SettingsView.js:39]: typo? ---> refreshing

File: src/config/config-template.js [hunk 7]
```diff
@@ -15,6 +15,7 @@ const config = {
 		diagramBackground: "rgba(224, 224, 224, 0.6)",
 		error: "#B71C1C",
 	},
+	textNumberOfEvents: "Number of events"
```
Comment (Munin33) [inline src/config/config-template.js]: missing ' , '

Comment (Munin33) [inline src/config/config-template.js]: can you move this to 'messages'?

File: src/index.js [hunk 8]
```diff
@@ -4,19 +4,19 @@ import App from './App';
 import {BrowserRouter as Router, Route, Switch, Redirect} from 'react-router-dom';
 import GridView from './GridView/GridView';
 import DetailView from './DetailView/DetailView';
-import EventTypesView from './EventTypesView/EventTypesView';
+import EventTypesView from './SettingsView/SettingsView';
```
Comment (Munin33) [inline src/index.js]: wrong import --> SettingsView

**Review by Munin33 — CHANGES_REQUESTED** [review]

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/a97e7f1c17255ac41b232908db4f4aee2316ba10/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/983b3aa6134e8cef056d9ac7377c0af8605b2b13/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/aa21063e8295c3afe749ad22020acdbb4ca747b5/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [EventQueryListItem.js#L20](https://github.com/bptlab/argos-frontend/blob/c785c9e8b8fa7a8d0bcd3f6a7230c6e2e01b068b/src/SettingsView/EventQueryListItem.js#L20): Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Review by Munin33 — APPROVED** [review]

**Pull request merged**